### PR TITLE
[PEPC-Boost][8] Block Assembler Client Integration tests

### PR DESCRIPTION
**Pull request by bharath-123**
## 📝 Summary

This PR implements tests for block assembler client integration with SubmitBlocks

### CODE REVIEW

File: services/api/pepc_boost_test.go [hunk 1]
```diff
@@ -13,18 +13,110 @@ import (
...
+		},
+		withdrawalsRoot: phase0.Root(withdrawalsRoot),
+	}
+	backend.relay.blockAssembler = &MockBlockAssembler{
+		assemblerError: nil,
+	}
+}
+
+func prepareBlockSubmitRequest(t *testing.T, payloadJSONFilename string, submissionSlot, submissionTimestamp uint64, backend *testBackend) *common.BuilderSubmitBlockRequest {
+	// Prepare the request payload
```
Comment (michaelneuder) [inline services/api/pepc_boost_test.go:62]: add t.Helper() to top of this fn

File: services/api/pepc_boost_test.go [hunk 2]
```diff
@@ -13,18 +13,110 @@ import (
...
+	// sign the payload with the builder keypair
+	signature, err := boosttypes.SignMessage(req.Message(), backend.relay.opts.EthNetDetails.DomainBuilder, secretKey)
+	require.NoError(t, err)
+	req.Capella.Signature = phase0.BLSSignature(signature)
+
+	return req
+}
+
+func assertTobTxs(t *testing.T, backend *testBackend, slot uint64, parentHash string, tobTxValue *big.Int, txHashRoot [32]byte) {
+	tobTxValue, err := backend.redis.GetTobTxValue(context.Background(), backend.redis.NewPipeline(), slot, parentHash)
```
Comment (michaelneuder) [inline services/api/pepc_boost_test.go:87]: add t.Helper() to top of fn

File: services/api/pepc_boost_test.go [hunk 3]
```diff
@@ -226,813 +318,718 @@ func TestCheckTxAndSenderValidity(t *testing.T) {
...
+			jsonReq, err = req.MarshalJSON()
+			require.NoError(t, err)
+			rr = backend.requestBytes(http.MethodPost, tobTxSubmitPath, jsonReq, map[string]string{
+				"Content-Type": "application/json",
+			})
+			if !c.nextSentIsHigher {
+				require.NoError(t, err)
+				require.Equal(t, http.StatusBadRequest, rr.Code)
+				require.Contains(t, rr.Body.String(), "TOB tx value is less than the current value!")
+			} else {
```
Comment (michaelneuder) [inline services/api/pepc_boost_test.go:478]: add return to if and then no need for the else

File: services/api/pepc_boost_test.go [hunk 4]
```diff
@@ -226,813 +318,718 @@ func TestCheckTxAndSenderValidity(t *testing.T) {
...
+			backend := newTestBackend(t, 1)
+
+			headSlot := uint64(32)
+			submissionSlot := headSlot + 1
+			submissionTimestamp := 1606824419
+
+			// Payload attributes
+			payloadJSONFilename := "../../testdata/submitBlockPayloadCapella_Goerli.json.gz"
+
+			parentHash := "0xbd3291854dc822b7ec585925cda0e18f06af28fa2886e15f52d52dd4b6f94ed6"
```
Comment (michaelneuder) [inline services/api/pepc_boost_test.go:819]: third time we have used these constants. can we def them at the top of the file ?

File: services/api/pepc_boost_test.go (hunk 1 above)
Comment (bharath-123) [reply services/api/pepc_boost_test.go:62]: done in a future PR!

File: services/api/pepc_boost_test.go (hunk 2 above)
Comment (bharath-123) [reply services/api/pepc_boost_test.go:87]: done in a future PR!

File: services/api/pepc_boost_test.go (hunk 3 above)
Comment (bharath-123) [reply services/api/pepc_boost_test.go:478]: got it!

File: services/api/pepc_boost_test.go (hunk 4 above)
Comment (bharath-123) [reply services/api/pepc_boost_test.go:819]: yes have done it in a future PR!

**Pull request merged**